apps/builder/app/dashboard/projects/thumbnail-link.tsx

```diff
@@ -30,10 +33,19 @@ const getThumbnailAbbreviation = (title: string) =>
 export const ThumbnailLink = forwardRef<
   HTMLAnchorElement,
   { title: string; to: string }
->(({ title, to }, ref) => (
-  <Link ref={ref} to={to} className={thumbnailStyle()} tabIndex={-1}>
-    {getThumbnailAbbreviation(title)}
-  </Link>
-));
+>(({ title, to }, ref) => {
+  const navigation = useNavigation();
+  const shouldShowLoading =
+    navigation.state !== "idle" && to === navigation.location.pathname;
+
+  return (
+    <Link ref={ref} to={to} className={thumbnailStyle()} tabIndex={-1}>
+      {shouldShowLoading ? (
+        <Loading type="svg" isDark={false} delay={700} />
```

Comment:
I would probably go lower with the delay, but not sure

Reply:
I will try to make it lower.